File: app/src/main/java/com/sstechcanada/todo/broadcast_receivers/DailyAlarmReceiver.java
package com.sstechcanada.todo.broadcast_receivers;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.sstechcanada.todo.services.DueCheckIntentService;

public class DailyAlarmReceiver extends BroadcastReceiver {
    public static final int REQUEST_CODE = 1201;
    public static final String ACTION = "com.sstechcanada.todo.alarm";

    @Override
    public void onReceive(Context context, Intent intent) {
        Intent dueCheckIntent = new Intent(context, DueCheckIntentService.class);
        dueCheckIntent.putExtra("foo", "bar");
        context.startService(dueCheckIntent);
    }
}

File: app/src/main/java/com/sstechcanada/todo/broadcast_receivers/DailyAlarmReceiver.kt
package com.sstechcanada.todo.broadcast_receivers

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.sstechcanada.todo.services.DueCheckIntentService

class DailyAlarmReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val dueCheckIntent = Intent(context, DueCheckIntentService::class.java)
        dueCheckIntent.putExtra("foo", "bar")
        context.startService(dueCheckIntent)
    }

    companion object {
        const val REQUEST_CODE = 1201
        const val ACTION = "com.sstechcanada.todo.alarm"
    }
}